File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
app.controller("brandController", function($scope,$controller,brandService) {
    $controller('baseController',{$scope:$scope});//继承
    //普通显示
    $scope.findAll = function(){
       brandService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }

    $scope.findPage = function (page,rows) {
       brandService.findPage(page,rows).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        )
    }
    //分页加模块查询显示数据
    $scope.searchEntity={};//定义搜索对象

    //搜索
    $scope.search=function(page,rows){
        brandService.search(page,rows,$scope.searchEntity).success(
            function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }

    $scope.save = function () {

        brandService.save($scope.entity).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                }else{
                    alert(response.message)
                }
            }
        )
    }

    $scope.findOne = function (id) {
       brandService.findOne(id).success(
            function (response) {
                $scope.entity = response;

            }
        )
    }


    $scope.del = function () {
       brandService.del($scope.selectIds).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                }else{
                    alert(response.message)
                }
            }
        )
    }

});

File: youlexuan-manager-web/src/main/java/com/youlexuan/manager/controller/BrandController.java
package com.youlexuan.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.youlexuan.entity.PageResult;
import com.youlexuan.entity.Result;
import com.youlexuan.pojo.TbBrand;
import com.youlexuan.sellergoods.service.BrandService;

import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(@RequestParam("page") int pageNum, @RequestParam("rows") int pageSize){
        return brandService.findPage(null,pageNum,pageSize);
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand tbBrand, @RequestParam("page") int pageNum, @RequestParam("rows") int pageSize){
        return brandService.findPage(tbBrand,pageNum,pageSize);
    }

    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand tbBrand){
        try {
            brandService.add(tbBrand);
            return  new Result(true,"添加成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"添加失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand tbBrand){
        try {
            brandService.update(tbBrand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("del")
    public Result del(Long[] ids){
        try {
            brandService.del(ids);
            return new Result(true,"删除成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("findBrand")
    public List<Map> findBrand(){
        return brandService.findBrand();
    }
}
